### PR TITLE
Don't have template engine crash on template errors

### PR DESCRIPTION
**Pull request by shanx**
Bind the template to the context to prevent: AttributeError: 'NoneType' object has no attribute 'engine'

### CODE REVIEW

File: mail_builder/builder.py [hunk 1]
```diff
@@ -28,21 +32,22 @@ def build_message(template_names, extra_context=None, force_multipart=False,
     data = dict(defaults)
     data.setdefault('body', '')
 
-    # Scalar values
```
Comment (funkybob) [inline mail_builder/builder.py:31]: This is good, and all, but breaks 1.7 compat... not that I'm all that bothered.

File: mail_builder/builder.py [hunk 2]
```diff
@@ -11,6 +11,9 @@ def load_template(template_names):
     # Unwrap for Django 1.8+
     template = getattr(template, 'template', template)
 
+    return template
+
+def get_blocks(template):
```
Comment (funkybob) [inline mail_builder/builder.py]: why did these functions need to be separated?

**Comment by funkybob** [comment]
I've inlined the template loading and block resolving, and merged in your fixes along wit that.

Thanks!

**Comment by shanx** [comment]
thanks for adapting and merging in the changes! sorry for breaking 1.7 compatibility! basically this fix is only required for 1.8 and up. Maybe it is also wise to add testing through tox so multiple versions can be tested? I'll add a tox.ini in a separate pull request

**Comment by funkybob** [comment]
I already use travis for testing, and am quite happy to drop 1.7 support, especially now on the cusp of the 1.10 release.

**Pull request closed without merging**